metaflow/event_logger.py

```diff
@@ -17,17 +20,32 @@ def terminate(self):
 
 
 class EventLogger(NullEventLogger):
-    def __init__(self, logger_type):
-        # type: (str) -> None
+    def __init__(self, logger_type, env, **kwargs):
+        # type: (str, str, str) -> None
         self.sidecar_process = None
         self.logger_type = logger_type
+        self._context = {"env": env.get_environment_info()}
+        # The additional keywords are added to the context for the monitor.
+        # One use case is adding tags like flow_name etc.
+        if kwargs:
+            # Make sure we copy it as we don't want it to change
+            self._context["user_context"] = dict(kwargs)
 
     def start(self):
-        self.sidecar_process = SidecarSubProcess(self.logger_type)
+        if self.sidecar_process is None:
+            self.sidecar_process = SidecarSubProcess(self.logger_type, self._context)
+
+    def add_context(self, **kwargs):
+        self._context["user_context"].update(kwargs)
+        if self.sidecar_process is not None:
+            msg = Message(MessageTypes.UPDATE_CONTEXT, self._context)
+            self.sidecar_process.send(msg)
```

Comment:
Can't this be something that the `KeyStoneLogger` can handle by itself at startup?

Reply:
The use case for this is to update the context after creation so it needs to be somewhat in the base class. Sidecars are free to ignore though. I am fine if the specific context (like environment) is in the subclass but the notion of a context should be base class stuff IMHO.